File: expectanim/src/main/java/com/github/florent37/expectanim/core/scale/ScaleAnimExpectationSameScaleAs.java
package com.github.florent37.expectanim.core.scale;

import android.view.View;

/**
 * Created by florentchampigny on 17/02/2017.
 */
public class ScaleAnimExpectationSameScaleAs extends ScaleAnimExpectationViewDependant {

    public ScaleAnimExpectationSameScaleAs(View otherView) {
        super(otherView, null, null);
    }

    @Override
    public Float getCalculatedValueScaleX(View viewToMove) {
        return otherView.getScaleX();
    }

    @Override
    public Float getCalculatedValueScaleY(View viewToMove) {
        return otherView.getScaleY();
    }
}

File: kotlinpleaseanimate/src/main/java/com/github/florent37/kotlin/pleaseanimate/core/scale/ScaleAnimExpectationSameScaleAs.kt
package com.github.florent37.kotlin.pleaseanimate.core.scale

import android.view.View

/**
 * Created by florentchampigny on 17/02/2017.
 */
class ScaleAnimExpectationSameScaleAs(otherView: View) : ScaleAnimExpectationViewDependant(otherView, null, null) {

    override fun getCalculatedValueScaleX(viewToMove: View): Float? {
        return otherView.scaleX
    }

    override fun getCalculatedValueScaleY(viewToMove: View): Float? {
        return otherView.scaleY
    }
}
